megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -555,11 +555,12 @@ private static ToHitData toHitCalc(Game game, int attackerId, Targetable target,
                             || (atype.getAmmoType() == AmmoType.T_NLRM)
                             || (atype.getAmmoType() == AmmoType.T_MEK_MORTAR))
                     && (munition.contains(AmmoType.Munitions.M_SEMIGUIDED))) {
-                for (TagInfo ti : game.getTagInfo()) {
-                    if (target.getId() == ti.target.getId()) {
-                        spotter = game.getEntity(ti.attackerId);
-                    }
-                }
+                final Targetable currentTarget = target; // Required for concurrency reasons
```

Comment:
For readability and code length, this does not feel like an improvement to me.

Reply:
`target` really should be final, but the current implementation for Swarm LRMs prevents that.

Reply:
I find this more readable because it makes it much easier to tell that it is simply getting any entity that successfully applied TAG to the target.

Reply:
If this is an issue, I could just drop this commit. I'd rather not hold up further review of this pull request due to a few controversial changes.

Reply:
This changes the code and may have other impacts as the old code picked the last target that matched where as the new code picks a random one.

Depending upon the rules, this may not be a valid change.

Reply:
This will literally never make a difference.